### PR TITLE
Closes #225 #232 #102 slight change to scope of package.  Removed onboarding temp…

### LINKED ISSUE #225: Documentation:  Reimagining the Onboarding Process
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

Hi @pharmaverse/admiral .  I wrote out some of my thoughts and what I heard from others on the onboarding process.  Essentially, I am seeing 4 Issues in our Onboarding Process that I would like to tackle in Q2.  I think Issue 2 looks particularly fun to create!  Please give feedback on the below if you think something is amiss or needing more cliarification and perhaps after Release time we can finalize and implement the changes.  

## Issue 1 
- [ ] Remove Onboarding Templates from Repos (No longer in use) -
 * - [ ] `admiraldev`, `admiral` and `admiraltemplate`
 * - [ ] Recommend other extension repos to remove as well. 
 
## Issue 2
- [ ] Explore "Dummy Issue" Idea for developers to try out and explore `admiral` process
 *   - [ ] How will this link into Contribution Model?
 *   - [ ] What type of issue? - Perhaps a simple `hello_admiral()` Feature Request with all the bells and whistles
 *   - [ ] Checklist for developers to walk through within the Dummy Issue for self-learning

## Issue 3
- [ ] Develop an above board process, but low maintenance, around brining in and onboarding for new core developers. 
*  - [ ]  Define loose criteria around commitment to `admiral` (e.g. 3 months of active work on code/discussion)
*  - [ ]  Give precedence to companies not represented on core team
*  - [ ]  Define how developers can reach out to us to initiate being brought into core team or how we reach out to you 
*  - [ ]  Define how one exits from core team (just don't whither away on us!)

## Issue 4 
- [ ] Develop a 10 minute highlight and welcoming video for folks wishing to be developers
*  - [ ]  Discontinuation of hello admiral  workshops (I think this is really cool, but seems like so much work!)
*  - [ ] Does this live on the youtube channel?

### PR DESCRIPTION
**Pull request by bms63**
…late

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi all.  Miscellaneous PR.  I tried to write a couple of simple tests for dev_utilities to make it interesting...

**Review by bundfussr — COMMENTED** [review]
Should we update the purpose as well?

File: tests/testthat/test-assertions.R [hunk 1]
```diff
@@ -65,8 +65,46 @@ test_that("assert_data_frame Test 4: error if not a dataframe", {
...
+      exprs(STUDYID, USUBJID),
+      extract_vars(order)
+    ))
+  }
+
+  expect_invisible(example_fun(input, order = exprs(SEQ)))
+})
+
+## Test 6: assert_data_frame throws not throw error if extract_vars() has NULL input ----
+test_that("assert_data_frame Test 6: assert_data_frame throws not throw error if extract_vars() has NULL input", { # nolint
```
Comment (zdz2101) [inline tests/testthat/test-assertions.R:88]:
```suggestion
test_that("assert_data_frame Test 6: assert_data_frame throws error if extract_vars() has NULL input", { # nolint
```

**Comment by bms63** [comment]
> Should we update the purpose as well?

Yes good call

File: tests/testthat/test-assertions.R [hunk 2]
```diff
@@ -65,8 +65,46 @@ test_that("assert_data_frame Test 4: error if not a dataframe", {
   )
 })
 
-## Test 5: error if dataframe is grouped ----
-test_that("assert_data_frame Test 5: error if dataframe is grouped", {
+## Test 5: assert_data_frame extract_vars() works as intended ----
+test_that("assert_data_frame Test 5: assert_data_frame extract_vars() works as intended", {
```
[the remark this answers is not in the document]
Comment (bms63) [reply tests/testthat/test-assertions.R:69]: ehh...I think I am going to leave as is.  There isn't an extract_vars section at the moment.

File: README.md [hunk 3]
```diff
@@ -26,13 +25,11 @@ The package is available from CRAN and can be installed by running
 To install the latest development version of the package directly from
 GitHub use the following code:
 
-```
-if (!requireNamespace("remotes", quietly = TRUE)) {
-  install.packages("remotes")
-}
+    if (!requireNamespace("remotes", quietly = TRUE)) {
```
Comment (zdz2101) [inline README.md:28]: not sure if the ticks were meant to be removed but the .rmd seems to be unchanged, unclear how this would render

**Review by zdz2101 — APPROVED** [review]
One small thing I think I missed on my first go-through but everything else looks good!

File: README.md (hunk 3 above)
Comment (bms63) [reply README.md:30]: I think it renders okay ??  
image
